### DISCUSSION TITLE
Flat orientation hexagon shape

### DISCUSSION THREAD

**Discussion by ishtanzar** [flauwekeul/honeycomb#89] [Q&A] [answered]
Hi,
I may be missing something but I can't figure out how to build an hexagon-shaped grid with flat orientation hexes:
```
const Hex = Honeycomb.extendHex({
            size: 1,
            orientation: 'flat'
        });
const hexGrid = Honeycomb.defineGrid(Hex);
const hexMap = hexGrid.hexagon({radius: 15});
```
I tried specifying the center with .hexagon({radius: 15, center: Hex(0)}) without better luck so far.
Note: I don't have this issue with a rectangle.
Any ideas ?
Thanks.

**Comment by flauwekeul** [comment, marked as answer]
That should work with v3. But it won't work with the beta of v4. I'm planning on releasing v4 stable soon, so I would advice you to use the beta now. The [spiral traverser](https://abbekeultjes.nl/honeycomb/guide/traversing-grids.html#spiral) is probably the best way to create a hexagon-shaped grid.

**Reply by ishtanzar** [reply]
Ok, thanks

**Reply by kirkbushell** [reply]
The problem with the spiral traverser is it's not the full grid, it's just a selection/highlight of the grid. Or is that the problem with v4 that I've been experiencing?

**Reply by flauwekeul** [reply]
You can use a traverser to [create a grid](https://abbekeultjes.nl/honeycomb/guide/creating-grids.html#new-grid-hex-traverser):

```ts
const Hex = defineHex()
const grid = new Grid(Hex, spiral({ start: [1, 2], radius: 2 }))
```

Or to [traverse an existing grid](https://abbekeultjes.nl/honeycomb/guide/traversing-grids.html#spiral):

```ts
const Hex = defineHex()
const grid = new Grid(Hex, rectangle({ with: 5, height: 5 }))

grid.traverse(spiral({ start: [1, 2], radius: 2 }))
```

**Reply by kirkbushell** [reply]
Thanks, mate. I found when I used this approach that the coordinates were a bit odd, and not what I needed (I need 0,0 to be the center of the hexagon).

I did end up figuring it out, though - ty very much for your response and help :)

**Comment by kirkbushell** [comment]
> Hi, I may be missing something but I can't figure out how to build an hexagon-shaped grid with flat orientation hexes:
> 
> ```
> const Hex = Honeycomb.extendHex({
>             size: 1,
>             orientation: 'flat'
>         });
> const hexGrid = Honeycomb.defineGrid(Hex);
> const hexMap = hexGrid.hexagon({radius: 15});
> ```
> 
> I tried specifying the center with .hexagon({radius: 15, center: Hex(0)}) without better luck so far. Note: I don't have this issue with a rectangle. Any ideas ? Thanks.

Where did you find docs on the hexagon? I've been wracking my brain trying to figure this out all night =\

**Reply by flauwekeul** [reply]
I hope [my reply](https://github.com/flauwekeul/honeycomb/discussions/89#discussioncomment-7049859) to your question helps you. If not: please let me know.
